client/src/components/campaignCreatorComponents/CampaignCreator.tsx

```diff
@@ -1,33 +1,40 @@
-import PillarOptions from './PillarOptions';
 import OptionListCampaignCreator from './OptionListCampaignCreator';
-import ccData from '../../static_data/campaign_creator.json';
 import RadioButtonListCampaignCreator from './RadioButtonListCampaignCreator';
 import { Button, Heading, Flex, Separator } from '@radix-ui/themes';
-import { TwilightSelect } from '../primitiveComponents/Primitives';
-import { useEffect } from 'react';
+import { TwilightSelect, TwilightTextField } from '../primitiveComponents/Primitives';
 import { useNavigate } from 'react-router-dom';
+import {
+    NemesisFightYearLists,
+    NodePillarLists,
+    TimelineOptionList,
+    TypeCampaignCreatorData,
+    TypeCampaignData,
+} from './CampaignTypeConfig';
+import presetCampaignData from '../../static_data/preset_campaigns.json';
+import campaignOptionsData from '../../static_data/campaign_creator.json';
 
 interface CampaignCreatorProps {
-    campaignSettings: Record<string, boolean>;
-    setCampaignSettings: React.Dispatch<React.SetStateAction<{}>>;
+    settlementName: string;
+    setSettlementName: React.Dispatch<React.SetStateAction<string>>;
+    selectedCampaign: TypeCampaignData;
+    setSelectedCampaign: React.Dispatch<React.SetStateAction<TypeCampaignData>>;
 }
 
-const CampaignCreator = ({ campaignSettings, setCampaignSettings }: CampaignCreatorProps) => {
+const CampaignCreator = ({
+    settlementName,
+    setSettlementName,
+    selectedCampaign,
+    setSelectedCampaign,
+}: CampaignCreatorProps) => {
     const navigate = useNavigate();
+    const selectCampaignOptions: string[] = Object.keys(presetCampaignData).sort();
 
-    const handleLoadPresetCampaignSelection = (campaign: string) => {
-        if (ccData.hasOwnProperty(campaign)) {
-            const modCampaignSettings: Record<string, boolean> = {
-                ...campaignSettings,
-            };
-            for (const key in modCampaignSettings) {
-                modCampaignSettings[key] = false;
-            }
-            const settings = ccData[campaign as keyof typeof ccData];
-            if (Array.isArray(settings)) {
-                settings.forEach((key: string) => (modCampaignSettings[key] = true));
-            }
-            setCampaignSettings(modCampaignSettings);
+    const handleSetPresetCampaign = (campaign: string) => {
+        const presetCampaigns: Record<string, TypeCampaignData> = { ...presetCampaignData };
+        if (presetCampaignData.hasOwnProperty(campaign)) {
```

Comment:
I have written `hasOwnProperty` maybe 4 times in my life in production code.  Don't validate your data.